pocs/scheduler/constraint.py

```diff
@@ -159,3 +159,29 @@ def get_score(self, time, observer, observation, **kwargs):
 
     def __str__(self):
         return "Moon Avoidance"
+
+
+class AlreadyVisited(BaseConstraint):
+
+    """ Simple Already Visited Constraint
+
+    A simple already visited constraint that determines if the given `observation`
+    has already been visited before. If given `observation` has already been
+    visited then it will not be considered for a call to become the `current observation`.
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def get_score(self, time, observer, observation, **kwargs):
+        veto = False
+        score = self._score
+
+        observed_list = kwargs.get('observed_list', observer.observed_list)
```

Comment:
I told you wrong here. `observed_list` belongs to the `scheduler`, not to the `observer` (the `observer` also belongs to the `scheduler`).  You can remove the `observer.observed_list` here and just fetch from `kwargs`. Then in the `dispatch.py` scheduler you can pass the `observed_list` in as a `common_properties` ([here](https://github.com/panoptes/POCS/blob/develop/pocs/scheduler/dispatch.py#L52)):

```
        common_properties = {
            'end_of_night': self.observer.tonight(time=time, horizon=-18 * u.degree)[-1],
            'moon': get_moon(time, self.observer.location),
            'observed_list': self.observed_list
        }
```

The `common_properties` get passed into as `kwargs` down below that.

Reply:
But this means we should add a test. You have added tests to successfully cover just the constraint, now we should add something in `test_dispatch_scheduler.py` that uses it as part of a normal scheduling routine.